jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdCredentials.java

```diff
@@ -137,12 +137,26 @@ private void validateClaims(OpenIdConfiguration configuration) throws Exception
             throw new AuthenticationException("Authorized party claim value should be the client_id");
 
         // Check that the ID token has not expired by checking the exp claim.
-        long expiry = (Long)claims.get("exp");
-        long currentTimeSeconds = (long)(System.currentTimeMillis() / 1000F);
-        if (currentTimeSeconds > expiry)
+        if (isExpired())
             throw new AuthenticationException("ID Token has expired");
     }
 
+    public boolean isExpired()
+    {
+        return checkExpiry(claims);
+    }
+
+    public static boolean checkExpiry(Map<String, Object> claims)
+    {
+        if (claims == null)
+            return true;
+
+        // Check that the ID token has not expired by checking the exp claim.
+        long expiry = (Long)claims.get("exp");
+        long currentTimeSeconds = System.currentTimeMillis() / 1000;
+        return (currentTimeSeconds > expiry);
```

Comment:
Let's try to avoid using `System.currentTimeMilllis()`, use:

```suggestion
        Instant.ofEpochSeconds((Long)claims.get("exp")).isBefore(Instant.now())
```

Please also fix `OpenIdCredentialsTest` by using `Instant.now().plusSeconds(5).getEpochSecond()`.